modules/xfeatures2d/src/daisy.cpp

```diff
@@ -0,0 +1,1754 @@
+
+    inline int quantize_radius( float rad ) const;
+
+    inline int filter_size( double sigma );
+
+    // Return a number in the range [-0.5, 0.5] that represents the location of
+    // the peak of a parabola passing through the 3 evenly spaced samples.  The
+    // center value is assumed to be greater than or equal to the other values
+    // if positive, or less than if negative.
+    inline float interpolate_peak( float left, float center, float right );
+
+
+}; // END DAISY_Impl CLASS
+
+
+// -------------------------------------------------
+/* DAISY computation routines */
+
+inline void DAISY_Impl::reset()
+{
+    m_image.release();
+
+    m_orientation_map.release();
+    m_scale_map.release();
+
+    m_dense_descriptors.release();
+    m_smoothed_gradient_layers.release();
+}
+
+inline void DAISY_Impl::release_auxiliary()
+{
+    m_orientation_map.release();
+    m_scale_map.release();
+
+    m_smoothed_gradient_layers.release();
+
+    m_grid_points.release();
+    m_oriented_grid_points.release();
+    m_cube_sigmas.release();
+
+    m_image.release();
+}
+
+// creates a 1D gaussian filter with N(mean,sigma).
+static void gaussian_1d( float* fltr, int fsz, float sigma, float mean )
+{
+    CV_Assert(fltr != NULL);
+
+    int sz = (fsz - 1) / 2;
+    int counter = -1;
+    float sum = 0.0f;
+    float v = 2 * sigma*sigma;
+    for( int x=-sz; x<=sz; x++ )
+    {
+      counter++;
+      fltr[counter] = exp((-((float)x-mean)*((float)x-mean))/v);
+      sum += fltr[counter];
+    }
+
+    if( sum != 0 )
+    for( int x=0; x<fsz; x++ ) fltr[x] /= sum;
+}
+
+// computes the gradient of an image
+static void gradient( Mat im, int h, int w, Mat dy, Mat dx )
+{
+    CV_Assert( !dx.empty() );
+    CV_Assert( !dy.empty() );
+
+    for( int y=0; y<h; y++ )
+    {
+      int yw = y*w;
+      for( int x=0; x<w; x++ )
+      {
+        int ind = yw+x;
+        // dx
+        if( x>0 && x<w-1 ) dx.at<float>(ind) = (im.at<float>(ind+1)-im.at<float>(ind-1)) / 2.0f;
+        if( x==0         ) dx.at<float>(ind) =  im.at<float>(ind+1)-im.at<float>(ind  );
+        if( x==w-1       ) dx.at<float>(ind) =  im.at<float>(ind  )-im.at<float>(ind-1);
+
+        // dy
+        if( y>0 && y<h-1 ) dy.at<float>(ind) = (im.at<float>(ind+w)-im.at<float>(ind-w)) / 2.0f;
+        if( y==0         ) dy.at<float>(ind) =  im.at<float>(ind+w)-im.at<float>(ind  );
+        if( y==h-1       ) dy.at<float>(ind) =  im.at<float>(ind  )-im.at<float>(ind-w);
+      }
+    }
+}
+
+static Mat layered_gradient( Mat data, int layer_no = 8 )
+{
+   int data_size = data.rows * data.cols;
+   Mat layers( 1, layer_no*data_size, CV_32F, Scalar(0) );
+
+   float kernel[5];
+   gaussian_1d(kernel, 5, 0.5f, 0.0f);
+   Mat Kernel(1, 5, CV_32F, (float*) kernel);
+
+   Mat cvO;
+   // smooth the data matrix
+   filter2D( data, cvO, CV_32F, Kernel, Point( -1, -1 ), 0, BORDER_REPLICATE );
```

Comment:
* As i mention it i was nto able to replace the two filter2D() with single sepFilter2D() without numeric impact in final descriptor. If doing it produced descriptors will not be numericaly 1:1 as original daisy.

 If numerical result slightly differ and this is acceptable here, then  yes i will replace.